### PR TITLE
3ds: support for 800x240 mode, minor improvements

### PR DESCRIPTION
**Pull request by asiekierka**
Adds *so much* legibility to the text.

Ready for review, but pending a new stable libctru release.

For future changelog notes: [According to libctru documentation,] this should work on all 3DS models excluding the "old" 2DS (but including the New 2DS XL).

### CODE REVIEW

**Comment by asiekierka** [comment]
New stable libctru/citro3d release is now a thing. As far as I know, no other changes to MegaZeux/3DS are necessary.

File: arch/3ds/Makefile.in [hunk 1]
```diff
@@ -62,6 +62,10 @@ arch/3ds/%.o: arch/3ds/%.c arch/3ds/shader_2d.shbin.o arch/3ds/shader_playfield.
 	$(if ${V},,@echo "  CC      " $<)
 	${CC} -MD ${core_cflags} ${core_flags} -Wno-unused-macros -c $< -o $@
 
+arch/3ds/%.o: arch/3ds/%.cpp
+	$(if ${V},,@echo "  CXX     " $<)
+	${CXX} -MD ${core_cxxflags} ${core_flags} -Wno-unused-macros -c $< -o $@
+
```
Comment (AliceLR) [inline arch/3ds/Makefile.in:68]: The shbin.o dependencies need to be moved from the .c rule to the .cpp rule. Parallel builds rely on that to complete properly.

**Comment by AliceLR** [comment]
I get a crash when switching between wide and regular mode when several layers are active. Aside from this and the Makefile issue I commented on above, everything looks good.

**Comment by asiekierka** [comment]
The crash is not related to layers; I managed to get it on the title screen of &, which is not an unbound game. The crash also seems to be unconnected to a high amount of changes performed, which (alongside a brief code audit) seem to rule out a memory leak. The crash is fairly inconsistent/rare, as well.

I have found a potential use-after-free in the relevant area of code which could be causing this, see https://github.com/devkitPro/citro3d/pull/52 - that might have been the reason.

**Comment by AliceLR** [comment]
I haven't been able to reproduce that crash using a build of citro3d with the aforementioned patch. It does seem to have fixed this feature. I'll merge this once there's an official update for citro3d (just in case of the unlikely event someone else tries to build this port without a manually patched citro3d...). The upper screen is readable enough with this that I think the help file could be enabled if you're still interested in that (either after this is merged or when I enable the editor in CONFIG.3DS for 2.92f).

**Pull request merged**